rfd/0177-upgrade-js-package-manager.md

```diff
@@ -0,0 +1,183 @@
+Let's go through installation options of the most popular package managers:
+* npm has `install-strategy`:
+    * hoisted (default): install non-duplicated in top-level, and duplicated as 
+  necessary within directory structure.
+    * nested: install in place, no hoisting. Note: this does not seem to work 
+  correctly https://github.com/npm/cli/issues/6537.
+    * shallow: only install direct deps at top-level. Note: it creates 
+  `node_modules` directory only in the project's root, so it does not change 
+  much when it comes to workspaces, the dependencies are still hoisted.
+    * linked: (experimental) install in node_modules/.store, link in place, 
+  unhoisted.
+  Note: workes like pnpm. See an [RFD](https://github.com/npm/rfcs/blob/main/accepted/0042-isolated-mode.md).
+* Yarn has `nodeLinker`: 
+    * pnp: no node_modules, unhoisted. Note: does not work with Electron.
+    * pnpm: install in `node_modules/.store`, link in place, unhoisted. 
+  Note: It should work with the latest alpha electron-builder and Electron 
+  versions.
+    * node_modules: flat, hoisted. Same as the node_modules created by npm or 
+  Yarn Classic.
+* pnpm has `node-linker`:
+    * isolated: dependencies are symlinked from a virtual store at
+  `node_modules/.pnpm`. Note: I was able to build Teleport Connect with the 
+  latest electron-builder and Electron versions.
+    * hoisted: a flat node_modules without symlinks is created. 
+  Same as the node_modules created by npm or Yarn Classic.
+    * pnp: no node_modules. Same as pnp in Yarn.
+
+The installation option that is most strict and works with Electron is 
+`linked`/`pnpm`. 
+
+### So, what package manager should we migrate to?
+Modern package managers offer similar features, so I think the strictness and 
+performance of the manager are the factors that differentiate them the most.
+Here pnpm appears to be the winner. It supports installing dependencies as 
+linked by default, which makes it the most battle-tested implementation.
+The speed is also great, it's consistently one of the fastest tools in 
+[benchmarks](https://p.datadoghq.eu/sb/d2wdprp9uki7gfks-c562c42f4dfd0ade4885690fa719c818?fromUser=false&refresh_mode=sliding&tpl_var_npm%5B0%5D=%2A&tpl_var_pnpm%5B0%5D=%2A&tpl_var_yarn-classic%5B0%5D=%2A&tpl_var_yarn-modern%5B0%5D=%2A&tpl_var_yarn-nm%5B0%5D=%2A&tpl_var_yarn-pnpm%5B0%5D=no&from_ts=1711447153431&to_ts=1719223153431&live=true).
+
+The npm support for `linked` is really interesting, but as long as it is 
+experimental, we should not use it in a production environment.
+
+When it comes to Yarn, the main advantage is that we already use it, so we could 
+still type `yarn start-teleport`. 
+However, if we are going to use pnpm installation method, why not switch to pnpm 
+entirely? 
+Yarn also shows a warning `The pnpm linker doesn't support providing different 
+versions to workspaces' peer dependencies` during the installation, but I'm not 
+sure what significance this has.
```

Comment:
@gzdunek It's not clear to me from this section what your actual recommendation is.

Reply:
(Oh, I noticed that it's explained below. I just expected it to be given in this particular section. :) )

Reply:
Thanks, I clarified that.